### PR TITLE
[Widgets screen] Fix categorization not showing in the global inserter

### LINKED ISSUE #26159: Inserter overcrowded with no categories
In the latest iteration of the Widget screen, the left sidebar inserter doesn't have any categories to easily find blocks to add to widget ares. After installing two heavy plugins (bbpress, LifterLMS) and a Genesis theme, the inserter doesn't feel helpful to use for the widget screen. 

Here is a video https://recordit.co/h0ZjHqHlCB

**To reproduce**
1. Install bbPress, LifterLMS
2. Go to the Widget Screen
3. Add bbpress login widget

**Expected behavior**
Some kind of order and accordion. At least an alphabetical order something. 

- WordPress version: 5.5.1
- Gutenberg Master Oct 15

**Desktop (please complete the following information):**
 - OS X 10.15.7,
 - Browser Chrome
 - Version 86.0.4240.75

### PR DESCRIPTION
**Pull request by kevin940726**
## Description

Fix #26159 and #26329.

Removed the now outdated check of `hasChildItems`.

## How has this been tested?

1. Go to the widgets screen.
2. Open the global inserter.
3. There should be categorization of blocks.

## Screenshots 
image

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by talldan** [comment]
@kevin940726 Before reviewing, I'd like to have some understanding of what the cause of this is. Why does this problem happen with the widget area block and not other blocks?

**Comment by kevin940726** [comment]
@talldan Sure thing! Sorry I didn't make it clear in the description.

It's because `widget-area` is itself a block which can has child blocks. When we're opening the global inserter in the widgets screen, we're effectively opening a child blocks inserter for the `widget-areas` block (defaults to the first widget area if not selected), which doesn't have categorization. Since `widget-area` is a special kind of block that we don't want it to open child blocks inserter, I'm adding a special flag for it to force showing the categorization in the inserter.

**Comment by talldan** [comment]
@kevin940726 The curious thing is that I don't see the same issue with other blocks like Group, Cover that are container blocks.

Looking at the code it seems to be because another block has its `parent` set to the widget area block:
https://github.com/WordPress/gutenberg/blob/master/packages/blocks/src/store/selectors.js#L194

That block is the legacy widget:
https://github.com/WordPress/gutenberg/blob/6b8bf9173b71a2bfc6679eb400cfd2776be1653d/packages/edit-widgets/src/blocks/legacy-widget/block.json#L29-L31

Legacy widget setting widget area as its parent shouldn't mean it's exclusive to the widget area, there can still be other blocks as children. I remember https://github.com/WordPress/gutenberg/pull/23231 changed how that works.

My conclusion is that this `hasChildItems` check is outdated since #23231 and can be removed/simplified. What do you think?

**Review by Onixmiha — APPROVED** [review]
packages/edit-widgets/src/blocks/widget-area/index.js

**Comment by kevin940726** [comment]
Ohh I didn't know that! I'll have to take a deeper look into this later.

**Comment by draganescu** [comment]
[Removing ](https://github.com/WordPress/gutenberg/commit/beec013a0ab629948cfbf3fa32b1eedc849ae157) this check seems to have no effect other than fixing the widgets editor. What should I test? With blocks installed via plugins the categories and collections displayed just fine.

**Comment by kevin940726** [comment]
> [Removing ](https://github.com/WordPress/gutenberg/commit/beec013a0ab629948cfbf3fa32b1eedc849ae157) this check seems to have no effect other than fixing the widgets editor. What should I test? With blocks installed via plugins the categories and collections displayed just fine.

Yeah, seems like we can safely remove it? I can add some e2e tests to both the post editor and the widgets editor.

**Review by draganescu — APPROVED** [review]
Looks like you fixed something via removal of something else, best fix. Ha ha :shipit:

File: packages/block-editor/src/components/inserter/block-types-tab.js [hunk 1]
```diff
@@ -94,56 +80,39 @@ export function BlockTypesTab( {
 
 	return (
 		<div>
-			{ hasChildItems && (
-				<ChildBlocks rootClientId={ rootClientId }>
```
Comment (talldan) [inline packages/block-editor/src/components/inserter/block-types-tab.js:98]: The `ChildBlocks` component seems unused and isn't exported publically, I think it'd be good to go ahead and remove it as well.

File: packages/block-editor/src/components/inserter/test/block-types-tab.js [hunk 2]
```diff
@@ -138,16 +135,14 @@ describe( 'InserterMenu', () => {
 		expect( blocks[ 2 ].textContent ).toBe( 'Some Other Block' );
 	} );
 
-	it( 'displays child blocks UI when root block has child blocks', () => {
-		useSelect.mockImplementation( () => true );
-
+	it( "doesn't display child blocks UI", () => {
```
Comment (talldan) [inline packages/block-editor/src/components/inserter/test/block-types-tab.js:138]: If we delete the `ChildBlocks` component, then I think we can also remove this test as well.

File: packages/block-editor/src/components/inserter/block-types-tab.js (hunk 1 above)
Comment (kevin940726) [reply packages/block-editor/src/components/inserter/block-types-tab.js:98]: Good catch!

**Review by talldan — APPROVED** [review]

**Pull request merged**